### ISSUE TITLE
Not compatible with protobufjs 6.7.x

### ISSUE THREAD

**Issue by JustGreg** [kondi/rxjs-grpc#7]
the rxjs-grpc cli produce this line:
```typescript
sayHello(request: (sample.HelloRequest|{ [k: string]: any })): Observable<sample.HelloReply>;
```
You will get the error `Property 'name' does not exist on type 'HelloRequest | { [k: string]: any; }'.`
If you change the pipe-symbol `|` to an ampersand-symbol `&` it should work.
```typescript
sayHello(request: (sample.HelloRequest & { [k: string]: any })): Observable<sample.HelloReply>;
```

**Comment by kondi** [comment]
Could you share your yarn lock file? Or if not used, the package json?

**Comment by JustGreg** [comment]
yarn.lock
[yarn.txt](https://github.com/kondi/rxjs-grpc/files/909559/yarn.txt)

**Comment by kondi** [comment]
Quick fix: revert back protobufjs to ~6.6.5.
Looks like it is not compatible currently with 6.7.x, I will look at it later.

**Comment by kondi** [comment]
You can update rxjs-grpc to 0.1.2, it has set the correct protobufjs version.

**Comment by JustGreg** [comment]
thank you, it works!

atom and vs code still show the error `Property 'xyz' does not exist...` with Typescript 2.2.1 and 2.2.2

**Comment by kondi** [comment]
It should work well in VS code, I am using it everyday. Can you share a bit more? At least the ts file and on which line you have the error.

**Comment by JustGreg** [comment]
**sample.proto**
```
syntax = "proto3";

package sample;

service Greeter {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc SayMultiHello (MultiHelloRequest) returns (stream HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The request message containing the user's name and how many greetings they want.
message MultiHelloRequest {
  string name = 1;
  required uint32 num_greetings = 2;
}

// The response message containing the greetings.
message HelloReply {
  string message = 1; 
}
```
**server.ts**
```typescript
import { Observable } from 'rxjs';
import { serverBuilder } from 'rxjs-grpc';
import { sample } from './grpc-namespaces';

async function main() {
  type ServerBuilder = sample.ServerBuilder;
  const server = serverBuilder<ServerBuilder>('sample.proto', 'sample');

  server.addGreeter({

    sayHello(request) {
      return Observable.of({
        message: 'Hello ' + request.name
      });
    },

    sayMultiHello(request) {
      return Observable.timer(100, 500)
        .mapTo({ message: `Hello ${request.name}` })
        .take(request.num_greetings);
    }

  });

  server.start('0.0.0.0:50051');
}

main().catch(error => console.error(error));
```
**client.ts**
```typescript
import { clientFactory } from 'rxjs-grpc';
import { sample } from './grpc-namespaces';

async function main() {
  type ClientFactory = sample.ClientFactory;
  const Services = clientFactory<ClientFactory>('sample.proto', 'sample');

  const services = new Services('localhost:50051');
  const greeter = services.getGreeter();

  await greeter.sayMultiHello({ name: 'world', num_greetings: 3 }).forEach(response => {
    console.log(`Multi greeting: ${response.message}`);
  });

  await greeter.sayHello({ name: 'world' }).forEach(response => {
    console.log(`Greeting: ${response.message}`);
  });

}

main().catch(error => console.error(error));
```

![server ts](https://cloud.githubusercontent.com/assets/2097571/24856548/68c48ad4-1de5-11e7-997c-8a2036b56761.png)

**Comment by JustGreg** [comment]
[yarn.lock](https://github.com/kondi/rxjs-grpc/files/909690/yarn.txt)

**Comment by kondi** [comment]
It looks like the grpc-namespaces is still the one generated with the old version.

**Comment by JustGreg** [comment]
you are right! i recreated the namespace-file and now everythink looks good

**Comment by kondi** [comment]
Released in [v0.2.0](https://github.com/kondi/rxjs-grpc/releases/tag/v0.2.0).
